### ISSUE TITLE
Unable to display model fields and custom form fields on Flask admin together

### ISSUE THREAD

**Issue by scottydelta** [flask-admin/flask-admin#1440]
Here is the code I am using:

    from ..models import phone_number

    class ProfileForm(wtf.Form):
        project_name = TextField('Project Name', validators=[validators.required()])
        email = EmailField('Email', [validators.DataRequired(), validators.Email()])
        input_file = FileField('Input File')
 
    class SettingModelView(AdminView):
        can_create = False
        can_delete = False
        form_create_rules = [rules.FieldSet(('ProfileForm.project_name', 'ProfileForm.email', 'phone_number'),'Fill Form')]
   
        def prefix_name(self, file_name):
            part_filename = base64.urlsafe_b64encode(file_name)
            return secure_filename('file-%s-%s.csv' % (part_filename, datetime.now().strftime("%Y-%m-%d-%H:%M:%S")))
 
        def __init__(self, model, session, *args, **kwargs):
            super(SettingModelView, self).__init__(model, session, *args, **kwargs)
            self.static_folder = 'static'
            self.endpoint = 'admin'
            self.name = 'Hey Matie'
        def on_model_change(self, form, model, is_created):
            pass
        @expose('/', methods=['POST', 'GET'])
        def index(self):
            form = ProfileForm(request.form)
            if form.validate_on_submit():
                 return "checked"
            return self.render(self.edit_template, form=form)`

**Comment by stemirkhan** [comment]
Please tell me, did you manage to solve this problem?

**Comment by jbbqqf** [comment]
Hi! Triaging older issues — this looks like a usage question rather than a bug. The reported issue (custom form replacing model fields) is the documented behavior of `ModelView.form`; the supported way to *add* extra non-model fields is `form_extra_fields`.

Evidence:
- `flask_admin.model.BaseModelView.form_extra_fields` is the documented hook for that exact use case — see [`flask_admin/model/base.py:763-786`](https://github.com/pallets-eco/flask-admin/blob/master/flask_admin/model/base.py#L763-L786): "Dictionary of additional fields. […] You can control order of form fields using `form_columns` property."
- It's also part of the public API reference: [`doc/api/mod_model.rst:52`](https://github.com/pallets-eco/flask-admin/blob/master/doc/api/mod_model.rst#L52) (`autoattribute:: form_extra_fields`).

In other words: keep the model fields as-is, set `form_extra_fields = {'phone_number': StringField('Phone'), 'input_file': FileField('Input File')}`, and use `form_columns` if you want a specific ordering. Setting `form = ProfileForm` replaces the scaffolded form entirely, which is the symptom seen in the original report.

If I'm wrong about the intent and there's still a flask-admin bug here, point me at a current-`master` reproducer and I'll dig further. Otherwise would you mind closing this one out?

---
*Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually.*
